### PR TITLE
archive demo user voter status updates

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

File: src/db_api_util.ts [hunk 1]
```diff
@@ -515,52 +515,37 @@ export async function getSlackThreadsForVoter(
   }
 }
 
-export async function archiveMessagesForDemoVoter(
-  userId: string,
-  twilioPhoneNumber: string
-): Promise<void> {
-  logger.info(`ENTERING DBAPIUTIL.getSlackThreadsForVoter`);
+export async function archiveDemoVoter(userId: string): Promise<void> {
+  logger.info(`ENTERING DBAPIUTIL.archiveDemoVoter`);
```
Comment (tomerovadia) [inline src/db_api_util.ts:522]: Can you add twilioPhoneNumber back? A unique voter is identified by their userId + twilioPhoneNumber throughout the code, so we should keep that consistent.

File: src/db_api_util.ts [hunk 2]
```diff
@@ -515,52 +515,37 @@ export async function getSlackThreadsForVoter(
...
-        AND user_id = $1
-        AND twilio_phone_number = $2`,
-      [userId, twilioPhoneNumber]
+        AND user_id = $1`,
+      [userId]
     );
 
     logger.info(
-      `DBAPIUTIL.archiveDemoVolunteerVoterClaims: Successfully cleared demo voter claims.`
+      `DBAPIUTIL.archiveDemoVoted: Successfully archived demo voter.`
```
Comment (tomerovadia) [inline src/db_api_util.ts]: typo (voted -> voter)

**Comment by tomerovadia** [comment]
The PR title says "archive demo user voter claims" but aren't those already archived and the change you're making is archiving voter status updates?

File: src/db_api_util.ts (hunk 1 above)
Comment (liewegas) [reply src/db_api_util.ts:522]: updated!

File: src/db_api_util.ts (hunk 2 above)
Comment (liewegas) [reply src/db_api_util.ts]: fixed

File: src/db_api_util.ts (hunk 1 above)
Comment (tomerovadia) [reply src/db_api_util.ts:522]: Are you sure that twilio_phone_number is not always set for voter_status_updates? I'm not seeing a place in the code where it's not set:

https://github.com/tomerovadia/VoterHelpLine/blob/f59d6d5cf8aa8799c44b8398dd91217251c0df70/src/router.ts#L98
https://github.com/tomerovadia/VoterHelpLine/blob/f59d6d5cf8aa8799c44b8398dd91217251c0df70/src/router.ts#L955
https://github.com/tomerovadia/VoterHelpLine/blob/f59d6d5cf8aa8799c44b8398dd91217251c0df70/src/slack_interaction_handler.ts#L162

and the production database shows no rows with it missing

Comment (liewegas) [reply src/db_api_util.ts:522]:
```
helpline=> select * from voter_status_updates ;
          created_at           |             user_id              | user_phone_number | twilio_phone_number | is_demo | voter_status | originating_slack_user_name | originating_slack_user_id | slack_channel_name | slack_channel_id | slack_parent_message_ts | action_ts | archived 
-------------------------------+----------------------------------+-------------------+---------------------+---------+--------------+-----------------------------+---------------------------+--------------------+------------------+-------------------------+-----------+----------
 2020-10-25 18:25:55.705072-05 | 0f28ee384ea7e2e1cf4dbfe3bc9f1fb2 | +19512883411      |                     | t       | UNKNOWN      |                             |                           |                    |                  |                         |           | t
 2020-10-25 18:26:24.972818-05 | 0f28ee384ea7e2e1cf4dbfe3bc9f1fb2 | +19512883411      |                     | t       | UNKNOWN      |                             |                           |                    |                  |                         |           | 
```

Comment (tomerovadia) [reply src/db_api_util.ts:522]: Is that in production? I'm not seeing any null

Screen Shot 2020-10-25 at 8 53 14 PM

I'm also not seeing rows that match that data in production

Comment (liewegas) [reply src/db_api_util.ts:522]: ah, there was a bug in the logInitialVoterStatus.  One instance in staging (manually fixed).

Comment (tomerovadia) [reply src/db_api_util.ts:522]: I'm terrified that we almost deployed that bug to production. Can you please double check to make sure everything else looks ready?

Comment (liewegas) [reply src/db_api_util.ts:522]: Checked again and it looks good to me. (Note that this was easy to fix since the field is redundant wrt is_demo)

**Review by tomerovadia — APPROVED** [review]

**Pull request merged**